File: src/app.controller.ts
import {All, Controller, Get, MethodNotAllowedException, Res} from "@nestjs/common";
import type {Response} from "express";
const version = "1.0.0";

@Controller()
export class RootController {
    @Get()
    public async index(@Res() response: Response): Promise<void> {
        response.status(200).json({message: "Stock Prices API", version});
    }

    @All()
    public methodNotAllowed(): never {
        throw new MethodNotAllowedException();
    }
}

File: src/app_controller.rs
use axum::routing::get;
use axum::{Json, Router};
use serde::Serialize;

use crate::error::method_not_allowed;

const VERSION: &str = env!("CARGO_PKG_VERSION");

#[derive(Debug, Serialize)]
pub struct ApiMetadata {
    message: &'static str,
    version: &'static str,
}

pub fn routes() -> Router {
    Router::new().route("/", get(index).fallback(method_not_allowed))
}

async fn index() -> Json<ApiMetadata> {
    Json(ApiMetadata {
        message: "Stock Prices API",
        version: VERSION,
    })
}
